### PR TITLE
21112023 sprint demo feedback

### PR DESCRIPTION
**Pull request by bulgarchik**

### CODE REVIEW

File: FastBank.ConsoleApplication/MenuOptions.cs [hunk 1]
```diff
@@ -225,14 +225,15 @@ public static void OpenManagerMenu()
                                     .Count();
 
             var menuOptions = $"{{{ActiveUser.Role}}} Welcome {ActiveUser.Name}\n" +
-                              $"\nYou have {messagesCount} new message{(messagesCount > 1 ? 's' : string.Empty)}\n" +
+                              $"\nYou have {messagesCount} new message{(messagesCount > 1 || messagesCount == 0 ? 's' : string.Empty)}\n" +
                               $"\nPlease choose your action: \n" +
-                              $"\n 1: Customer transactions report" +
+                              $"\n 1: Customers transactions" +
                               $"\n 2: Manage messages" +
                               $"\n 3: Empoyees" +
```
Comment (AngelovGrowth) [inline FastBank.ConsoleApplication/MenuOptions.cs:232]: Empoyees -> Employees

File: FastBank.ConsoleApplication/MenuOptions.cs [hunk 2]
```diff
@@ -330,9 +336,14 @@ static public void OpenCustomerMenu()
                 var menuOptions = $"{{{ActiveUser.Role}}} Welcome {ActiveUser.Name}\n" + 
                                   $"\nYou bank amount: {customerBankAccount.Amount:0.00} \n" +
                                   $"\nPlease choose your action: \n" +
-                                  $"\n 1: For deposit \n 2: For withdraw \n 3: Create inquiry " +
-                                    $"\n 4: Check inquiries \n 5: Transfer to friend \n 0: Exit";
-                var commandsCount = 6;
+                                  $"\n 1: For deposit " +
```
Comment (AngelovGrowth) [inline FastBank.ConsoleApplication/MenuOptions.cs:339]: I would like globally to remove that "For" in each menu option

File: FastBank.Infrastructure/Repository/TransactionRepository.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ public class TransactionRepository : ITransactionRepository
     {
         private readonly IRepository _repository;
 
-        public const int TransactionPerPage = 10;
+        public const int TRANSACTION_PER_PAGE = 3;
```
Comment (AngelovGrowth) [inline FastBank.Infrastructure/Repository/TransactionRepository.cs:12]: TRANSACTIONS (Plural), not TRANSACTION_PER_PAGE. Should be TRANSACTIONS_PER_PAGE = 3

File: FastBank.Infrastructure/Repository/TransactionRepository.cs [hunk 4]
```diff
@@ -42,6 +42,32 @@ public int GetCustomersTransactionsCount()
...
         }
 
+        public int GetCustomerTransactionsCount(User user)
+        {
+            var transactionsCount = _repository.Set<TransactionDTO>()
+                .Include(t => t.CreatedByUser)
+                .Where(t => t.CreatedByUser.Role == Role.Customer.ToString() && t.CreatedByUser.UserId == user.Id)
+                .OrderBy(t => t.CreatedDate)
+                .Select(t => t.ToDomainObj())
+                .Count();
```
Comment (AngelovGrowth) [inline FastBank.Infrastructure/Repository/TransactionRepository.cs:52]: Why you need order by in that query when you return single int?

File: FastBank.Infrastructure/Repository/TransactionRepository.cs [hunk 5]
```diff
@@ -56,5 +82,51 @@ public void AddTransactionsReport(TransactionsReport transactionsReport)
         {
             _repository.Add<TransactionsReportDTO>(new TransactionsReportDTO(transactionsReport));
         }
+
+        public List<TransactionsReport> GetTransactionsReports(int currentPage = 1)
```
Comment (AngelovGrowth) [inline FastBank.Infrastructure/Repository/TransactionRepository.cs:86]: The class name is not ok. Should be TransactionReport not TransactionReports. TransactionReports for me means that it holds in a property a list of TransactionReport, but this is not true. Also method name should be changed accordingly.

Comment (AngelovGrowth) [reply FastBank.Infrastructure/Repository/TransactionRepository.cs:86]: Also check and rename the variables in that method

File: FastBank.Infrastructure/Repository/TransactionRepository.cs [hunk 6]
```diff
@@ -56,5 +82,51 @@ public void AddTransactionsReport(TransactionsReport transactionsReport)
...
+
+            foreach (var item in transactionsReports)
+            {
+                item.Index = currentPageTransactionsReportsIndex++;
+            }
+
+            return transactionsReports;
+        }
+
+        public List<TransactionsReport> GetUserLastTransactionsReports(User user, int count = 3)
```
Comment (AngelovGrowth) [inline FastBank.Infrastructure/Repository/TransactionRepository.cs:106]: Rename as above and check variable names

File: FastBank.Services/TransactionService/TransactionService.cs [hunk 7]
```diff
@@ -70,30 +295,20 @@ public void TransactionReport(User user)
...
-                        $"| {transaction.Amount}",
-                        $"| {transaction.TransactionType.GetDisplayName()}",
-                        $"| {transaction.CreatedByUser.Email}");
-                }
+                ShowTransactionList(transactions);
 
                 Console.WriteLine($"\nPage {currentPage}/{totalPages}\n");
 
+                var transactionsReports = ShowLastTransactionsReports(user);
+
```
Comment (AngelovGrowth) [inline FastBank.Services/TransactionService/TransactionService.cs:303]: Fix variable and method name as mentioned how i want them to be named in above comments

File: FastBank.Domain/TransactionsReport.cs [hunk 8]
```diff
@@ -13,5 +13,6 @@ public TransactionsReport(Guid reportId, DateTime createdOn, string pathToFile,
         public DateTime CreatedOn { get; private set; }
         public string PathToFile { get; private set; }
         public User CreatedBy { get; private set; }
+        public int Index { get; set; }
```
Comment (AngelovGrowth) [inline FastBank.Domain/TransactionsReport.cs:16]: Write new method SetIndex() to set the index of the report. Make the setter private.

File: FastBank.Infrastructure/Repository/TransactionRepository.cs (hunk 5 above)
Comment (bulgarchik) [reply FastBank.Infrastructure/Repository/TransactionRepository.cs:86]: Not agree with comments. The class is for Report which can have few or more transactions

File: FastBank.Domain/RepositoryInterfaces/ITransactionRepository.cs [hunk 9]
```diff
@@ -4,22 +4,22 @@ public interface ITransactionRepository
...
 
-        public int GetCustomersTransactionsCount();
+        public int GetCustomerTransactionsCount();
 
         public void AddTransaction(Transaction transaction);
 
         public void AddTransactionOrder(TransactionOrder transactionOrder);
 
-        public void AddTransactionsReport(TransactionsReport transactionsReport);
+        public void AddTransactionsFileReport(TransactionsFileReport transactionsReport);
```
Comment (AngelovGrowth) [inline FastBank.Domain/RepositoryInterfaces/ITransactionRepository.cs:13]: transactionsReport - transactionsFileReport

File: FastBank.Infrastructure/DTOs/TransactionsFileReportDTO.cs [hunk 10]
```diff
@@ -5,10 +5,10 @@
 namespace FastBank.Infrastructure.DTOs
 {
     [Table("TransactionsReports")]
-    public class TransactionsReportDTO
+    public class TransactionsFileReportDTO
     {
-        private TransactionsReportDTO() { }
-        public TransactionsReportDTO(TransactionsReport transactionsReport)
+        private TransactionsFileReportDTO() { }
+        public TransactionsFileReportDTO(TransactionsFileReport transactionsReport)
```
Comment (AngelovGrowth) [inline FastBank.Infrastructure/DTOs/TransactionsFileReportDTO.cs:11]: transactionsReport - transactionsFileReport

**Comment by AngelovGrowth** [comment]
@bulgarchik After resolving my last comments, changes are ready to be merged into master branch.

**Pull request merged**